src/main/java/edu/kis/powp/jobs2d/command/gui/CommandEditorWindow.java

```diff
@@ -0,0 +1,150 @@
+        container.add(coordY);
+        content.add(container, c);
+
+        c.gridy = 5;
+        c.fill = GridBagConstraints.BOTH;
+        changeCoordinates = new JButton("Change coordinates ");
+        changeCoordinates.setPreferredSize(new Dimension(300, 30));
+        changeCoordinates.addActionListener((ActionEvent e) -> this.changeCoordinates(Integer.parseInt(coordX.getText()), Integer.parseInt(coordY.getText()),commandPartsList.getSelectedValue()));
+        content.add(changeCoordinates, c);
+    }
+
+    public void changeCoordinates(int x, int y, DriverCommand command) {
+        if(commandManager.getCurrentCommand() instanceof ICompoundCommand) {
+            if (command instanceof SetPositionCommand) {
+                commandManager.setCurrentCommand(
+                        ((ICompoundCommand) commandManager.getCurrentCommand()).changeCoordinates(command, new SetPositionCommand(x, y)));
+            } else if (command instanceof OperateToCommand) {
+                commandManager.setCurrentCommand(
+                        ((ICompoundCommand) commandManager.getCurrentCommand()).changeCoordinates(command, new OperateToCommand(x, y)));
+            }
+        }
+    }
+
+    public void moveUp(DriverCommand command) {
+        if (commandManager.getCurrentCommand() instanceof ICompoundCommand) {
+            commandManager.setCurrentCommand(((ICompoundCommand) commandManager.getCurrentCommand()).moveUpCommand(command));
+        }
+        updateCurrentCommandField();
+    }
+
+    public void moveDown(DriverCommand command) {
+        if (commandManager.getCurrentCommand() instanceof ICompoundCommand) {
+            commandManager.setCurrentCommand(((ICompoundCommand) commandManager.getCurrentCommand()).moveDownCommand(command));
+        }
+        updateCurrentCommandField();
+    }
+
+    @Override
+    public void HideIfVisibleAndShowIfHidden() {
+        if (this.isVisible()) {
+            this.setVisible(false);
+        } else {
+            this.setVisible(true);
+        }
+    }
+
+    public void updateCurrentCommandField() {
+        currentCommandField.setText(commandManager.getCurrentCommandString());
+    }
+
+    public void updateCurrentCommandPartList() {
+        DriverCommand command = commandManager.getCurrentCommand();
+        DefaultListModel<DriverCommand> listModel = (DefaultListModel<DriverCommand>) commandPartsList.getModel();
+        listModel.removeAllElements();
+        addCommandsToList(listModel, command);
+    }
+
+    private void addCommandsToList(DefaultListModel<DriverCommand> listModel, DriverCommand command) {
+        if (command instanceof ICompoundCommand) {
+            ((ICompoundCommand) command).iterator().forEachRemaining(cmd -> {
+                if (cmd instanceof ICompoundCommand) {
+                    addCommandsToList(listModel, cmd);
+                } else {
+                    listModel.addElement(cmd);
+                }
+            });
+        }
```

Comment:
The same issue as above. Maybe instanceof can be avoided by making sure of that the method is working on mutable commands instances (not DriverCommand in general)?